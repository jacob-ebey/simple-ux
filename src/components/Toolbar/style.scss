@import '../variables';
@import '../Content/style';
@import '../Space/style';

.toolbar {
  background-color: $color-primary;

  &-body {
    @extend .content;
    @extend .spacer;
    overflow-x: auto;
  }


  &-items, &-items>*, &-title, &-title>* {
    color: $color-light;
    white-space: nowrap;
  }

  &-title {
    font-size: 1.2em;
    font-weight: 600;
    white-space: nowrap;

    &>a, &>a:hover {
      border-bottom: none;
    }
  }

  &-items {
    margin-left: 1em;
  }

  &-items>* {
    margin: 0 0.5em;
  }
}