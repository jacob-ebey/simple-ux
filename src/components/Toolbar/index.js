import { h } from 'preact' // eslint-disable-line no-unused-vars

import { merge, childArray } from '../tools'

import './style.scss'

const Toolbar = ({ title, class: className, children, ...rest }) => {
  const c = childArray(children)

  return (
    <nav
      {...rest}
      class={merge('toolbar', className)}
    >
      <div class='toolbar-body'>
        { title && <div class='toolbar-title'>{title}</div>}
        { c.length > 0 && <div class='toolbar-items'>{children}</div> }
      </div>
    </nav>
  )
}

export default Toolbar
