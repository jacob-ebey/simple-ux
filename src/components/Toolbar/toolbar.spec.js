/* eslint-env jest */
import { h } from 'preact' // eslint-disable-line no-unused-vars
import { shallow } from 'preact-render-spy'

import Toolbar from './'

describe('Toolbar', () => {
  describe('- snapshots -', () => {
    it('can render', () => {
      const wrapper = shallow(<Toolbar />)
      expect(wrapper).toMatchSnapshot()
    })

    it('can add custom class', () => {
      const wrapper = shallow(<Toolbar class='custom' />)
      expect(wrapper).toMatchSnapshot()
    })

    it('can render children', () => {
      const wrapper = shallow(
        <Toolbar>
          <p>Child 1</p>
          <p>Child 2</p>
        </Toolbar>
      )
      expect(wrapper).toMatchSnapshot()
    })

    it('can render title', () => {
      const wrapper = shallow(<Toolbar title={<a>Title</a>} />)
      expect(wrapper).toMatchSnapshot()
    })
  })
})
