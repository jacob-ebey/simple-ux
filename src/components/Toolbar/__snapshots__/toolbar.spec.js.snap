// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Toolbar - snapshots - can add custom class 1`] = `
preact-render-spy (1 nodes)
-------
<nav class="toolbar custom">
  <div class="toolbar-body"></div>
</nav>

`;

exports[`Toolbar - snapshots - can render 1`] = `
preact-render-spy (1 nodes)
-------
<nav class="toolbar">
  <div class="toolbar-body"></div>
</nav>

`;

exports[`Toolbar - snapshots - can render children 1`] = `
preact-render-spy (1 nodes)
-------
<nav class="toolbar">
  <div class="toolbar-body">
    <div class="toolbar-items">
      <p>Child 1</p>
      <p>Child 2</p>
    </div>
  </div>
</nav>

`;

exports[`Toolbar - snapshots - can render title 1`] = `
preact-render-spy (1 nodes)
-------
<nav class="toolbar">
  <div class="toolbar-body">
    <div class="toolbar-title">
      <a>Title</a>
    </div>
  </div>
</nav>

`;
